### PR TITLE
buildahimage: package install fails without containers-selinux

### PR DESCRIPTION
**Pull request by vbatts**
https://quay.io/repository/openshift-pipeline/buildah/build/37fc6d86-21d5-4e7b-b79c-44f9c55ca233

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

**Comment by vbatts** [comment]
the error is in the `yum install` line, which is swallowed by the ';', so i've added another commit to fix that up as well.

**Comment by rhatdan** [comment]
@vbatts Where does the install fail?  container-selinux should be Recommends?  Is this a problem with some other package requiring it?

container-selinux sucks in all of selinux-policy which is very large and can cause Issues confusing the container about whether SELinux is enabled or not.

**Comment by vbatts** [comment]
@rhatdan expand the "RUN" step where `yum install` is, and you'll see the error. 
```shell
STEP 1: FROM fedora
STEP 2: RUN yum -y install buildah --exclude container-selinux; rm -rf /var/cache /var/log/dnf* /var/log/yum.*
Fedora Modular 29 - x86_64                      783 kB/s | 1.5 MB     00:01
Fedora Modular 29 - x86_64 - Updates             34 kB/s | 2.1 MB     01:03
Fedora 29 - x86_64 - Updates                    1.9 MB/s |  25 MB     00:12
Fedora 29 - x86_64                              1.4 MB/s |  62 MB     00:45
Last metadata expiration check: 0:00:01 ago on Wed Apr 17 13:03:00 2019.
Error:
 Problem: conflicting requests
  - package buildah-1.4-1.dev.git0a7389c.fc29.x86_64 requires runc >= 1.0.0-17, but none of the providers can be installed
  - package buildah-1.7-3.git873f001.fc29.x86_64 requires runc >= 1.0.0-17, but none of the providers can be installed
  - package runc-2:1.0.0-68.dev.git6635b4f.fc29.x86_64 requires container-selinux >= 2:2.2-2, but none of the providers can be installed
  - package runc-2:1.0.0-55.dev.git578fe65.fc29.x86_64 requires container-selinux >= 2:2.2-2, but none of the providers can be installed
  - package container-selinux-2:2.69-3.git452b90d.module_2040+0e96cf1b.noarch is excluded
  - package container-selinux-2:2.69-3.git452b90d.module_2106+3b446c4f.noarch is excluded
  - package runc-2:1.0.0-50.dev.git20aff4f.module_2040+0e96cf1b.x86_64 is excluded
  - package runc-2:1.0.0-50.dev.git20aff4f.module_2106+3b446c4f.x86_64 is excluded
  - package container-selinux-2:2.95-1.gite3ebc68.fc29.noarch is excluded
  - package container-selinux-2:2.73-2.gitd7a3f33.fc29.noarch is excluded
(try to add '--skip-broken' to skip uninstallable packages)
--> 679d52ae50db39f9f20a47cf67fd3b2e89b52ae9c6756b007f02738951e49f60
STEP 3: FROM 679d52ae50db39f9f20a47cf67fd3b2e89b52ae9c6756b007f02738951e49f60
STEP 4: RUN sed 's|^#mount_program|mount_program|g' -i /etc/containers/storage.conf
sed: can't read /etc/containers/storage.conf: No such file or directory
Error: error building at step {Env:[PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin DISTTAG=fcontainer FGC=f FBR=f] Command:run Args:[sed 's|^#mount_program|mount_program|g' -i /etc/containers/storage.conf] Flags:[] Attrs:map[] Message:RUN sed 's|^#mount_program|mount_program|g' -i /etc/containers/storage.conf Original:RUN sed 's|^#mount_program|mount_program|g' -i /etc/containers/storage.conf}: error while running runtime: exit status 2
```

**Comment by vbatts** [comment]
if this is a fix elsewhere in the packaging, so be it, but we're setting up for daily builds of this as a container to `quay.io/openshift-pipeline/buildah`

**Comment by vbatts** [comment]
for example, i've just kicked off builds of the these buildahimages
https://quay.io/repository/openshift-pipeline/buildah?tab=builds

The `stable`(`latest`) will be built weekly. The `testing` is built daily. And `git` is built every four hours.

**Comment by vbatts** [comment]
@rhatdan ... it looks like the `testing` image _does_ install correctly.. so this seems to be a packaging issue with what is in the fedora stable repos

**Comment by rhatdan** [comment]
This issue is in the version of runc.
I just pushed an updated version of runc to stable.  Got lost in the shuffle.

**Comment by rhatdan** [comment]
* Mon Mar 11 2019 Dan Walsh (Bot) <dwalsh@fedoraproject.org> - 2:1.0.0-82.dev.git2b18fe1
- Change Requires container-selinux to recommends container-selinux

File: buildahimage/stable/Dockerfile [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
+# used to create a secured container that runs safely with privileges within
+# the container.
 #
 FROM fedora
 
-# Don't include container-selinux and remove 
-# directories used by dnf that are just taking
-# up space.
-RUN yum -y install buildah --exclude container-selinux; rm -rf /var/cache /var/log/dnf* /var/log/yum.*
+# Install build and remove directories used by dnf that are just taking up
```
Comment (TomSweeneyRedHat) [inline buildahimage/stable/Dockerfile]: s/build/Bulidah/

File: buildahimage/stable/Dockerfile [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
+# space.
+RUN yum -y install buildah && rm -rf /var/cache /var/log/dnf* /var/log/yum.*
```
Comment (TomSweeneyRedHat) [inline buildahimage/stable/Dockerfile]: I'll leave it to you and @rhatdan  to thumb wrestle this change

**Comment by vbatts** [comment]
oh we can close this and just fix the rpm to not fail on installing without containers-selinux

**Comment by rhatdan** [comment]
runc version should be showing up soon, although I still see it in updates-testing.

**Comment by rhatdan** [comment]
I just tried this with fedora29 and it successfully installs with the new runc in stable.

**Comment by vbatts** [comment]
sure enough. I just kicked the builds (https://quay.io/repository/openshift-pipeline/buildah?tab=builds) and stable built fine, is smaller (99.8mb) and can be pulled with `podman pull quay.io/openshift-pipeline/buildah`

**Pull request closed without merging**